extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/java/io/quarkus/resteasy/reactive/server/runtime/ResteasyReactiveRecorder.java

```diff
@@ -225,4 +228,42 @@ public ServerRestHandler get() {
         }
     }
 
+    private static class MultipartHandlerSupplier implements Supplier<ServerRestHandler> {
+
+        private volatile Boolean handleFileUploads;
+        private volatile String uploadsDirectory;
+        private volatile Boolean mergeFormAttributes;
+        private volatile Boolean deleteUploadedFilesOnEnd;
+        private volatile Long maxBodySize;
+
+        @Override
+        public ServerRestHandler get() {
+            if (handleFileUploads == null) {
+                synchronized (this) {
+                    if (handleFileUploads == null) {
+                        // read the config like this in order to be able to keep ResteasyReactiveRecorder#createDeployment
+                        // as a static-init recorder
+                        // TODO: reevaluate this
+                        handleFileUploads = ConfigProvider.getConfig().getOptionalValue("quarkus.http.body.handle-file-uploads",
+                                Boolean.class).orElse(true);
+                        uploadsDirectory = ConfigProvider.getConfig().getOptionalValue("quarkus.http.body.uploads-directory",
```

Comment:
I think there are a few problems here:
- This is runtime config being read in static init, so it will appear to work in JVM mode but won't work as expected in native mode.
- Currently this config only applies to the Vert.x body handler, which RESTEasy Reactive does not use. I don't know if we really want to overload this configuration.
- handle-file-uploads is not a good config option imho, the fact that the user has defined a multipart resource that is expecting a file is an indication that they want to handle file uploads. We should accept files on multipart endpoints, and for other endpoints leave it up to the MessageBodyReader how to handle it. 
- The default upload directory should be System.getProperty("java.io.tmpdir"), not 'file-uploads', as some apps won't have permission to write to the current working directory.
- merge-form-attributes makes no sense in this context, it just adds the form params to the query params, so will cause problems for us if you have a form param and a query param of the same name (basically you will be able to use @QueryParam to inject what should be an @FormParam). 
- The max request size is available under io.quarkus.vertx.http.runtime.VertxHttpRecorder#MAX_REQUEST_SIZE_KEY, and for fixed length requests is already enforced. I think our impl should just use this to avoid the static init config issue.

Reply:
The problem with `quarkus.http.body.uploads-directory` I think is that because of its name, users will expect it to work properly for all HTTP stacks. I could certainly remove it from here, but wouldn't it be confusing for users that such a "global" HTTP property doesn't work?
About `quarkus.http.body.handle-file-uploads`, I certainly agree but again we have the problem that "global" HTTP configuration will appear to not work in RESTEasy Reactive. Are we OK with that?

Reply:
I overhauled how this configuration is passed and removed the fileds you mentioned that are non necessary.

However, I haven't changed the default values and still rely on Quarkus defaults.
Let me know if you still think we should not abide by the Quarkus "global" http settings

Reply:
FTR, I do agree that an app should have a single "tmp" folder where it puts stuff. If that's configured with `java.io.tmpdir` then fair enough, but perhaps `quarkus.http.body.uploads-directory` should default to `{java.io.tmpdir}/uploads` then?

Reply:
That makes a lot of sense. Let me see if Quarkus config allows that

Reply:
OK, I changed the default to use the tmp dir approach.

cc @gsmet @cescoffier

Reply:
@stuartwdouglas I think I addressed all of your concerns